docs/migration/2211_35/modernize-apps-migrated-from-6.8-to-2211.19.md

```diff
@@ -0,0 +1,430 @@
++ import { dirname, join, resolve } from 'node:path';
++ import { fileURLToPath } from 'node:url';
++ import AppServerModule from './src/main.server';
+```
+
+2. Replace two old constants: `distFolder` and `indexHtml` with new three constants: `serverDistFolder`, `browserDistFolder`, `indexHtml`, according to the diff below:
+
+```diff
+- const distFolder = join(process.cwd(), 'dist/YOUR-APP-NAME/browser');
+- const indexHtml = existsSync(join(distFolder, 'index.original.html'))
+-   ? 'index.original.html'
+-   : 'index';
+
++  const serverDistFolder = dirname(fileURLToPath(import.meta.url));
++  const browserDistFolder = resolve(serverDistFolder, '../browser');
++  const indexHtml = join(browserDistFolder, 'index.html');
+```
+
+3. In the call `server.set('views, `... please use the constant `browserDistFolder` instead of `distFolder`  as a second argument.
+
+```diff
+-  server.set('views', distFolder);
++  server.set('views', browserDistFolder);
+```
+
+4. In the call `express.static( `... please use the constant `browserDistFolder` instead of `distFolder` as an argument.
+
+```diff
+   server.get(
+     '*.*',
+-    express.static(distFolder, {
++    express.static(browserDistFolder, {
+```
+
+5. Remove the block of code in the bottom of the file related handling Webpack's `require`, but leave only the call of the function `run()`
+
+```diff
+- // Webpack will replace 'require' with - '__webpack_require__'
+- // '__non_webpack_require__' is a proxy to Node 'require'
+- // The below code is to ensure that the server is run - only when not requiring the bundle.
+- declare const __non_webpack_require__: NodeRequire;
+- const mainModule = __non_webpack_require__.main;
+- const moduleFilename = (mainModule && mainModule.- filename) || '';
+- if (moduleFilename === __filename || moduleFilename.- includes('iisnode')) {
+-   run();
+- }
+
++ run();
+```
+
+6. In the very bottom of the file, remove the re-export of the path `./src/main.server`
+
+```diff
+- export * from './src/main.server';
+```
+
+This is the end of the migration guide for the Angular `application` builder.
+
+## In `app.module.ts`, use new, non-deprecated Angular APIs
+
+Unrelated to the migration to the new Angular `application` builder, please also update the `app.module.ts` file to use new, non-deprecated Angular APIs:
+
+### `src/app/app.module.ts`
+
+1. Remove the `HttpClientModule` from the `imports` array. 
+
+```diff
+  imports: [
+-    HttpClientModule,
+```
+
+2. Add `provideHttpClient(withFetch(), withInterceptorsFromDi()),` to the `providers` array.
+
+```diff
+  providers: [
++   provideHttpClient(withFetch(), withInterceptorsFromDi()),
+  ],
+```
+
+## For SSR projects, additionally:
+
+### `src/app/app.module.ts`
+
+Replace the item in the `imports` array `BrowserModule.withServerTransition({ appId: 'serverApp' }),` with just `BrowserModule`
+
+```diff
+  imports: [
+-    BrowserModule.withServerTransition({ appId: 'serverApp' }),
++    BrowserModule,
+  ],
+```
```

Comment:
```suggestion
```diff
  imports: [
-    BrowserModule.withServerTransition({ appId: 'serverApp' }),
+    BrowserModule,
  ],
  /* ... */
  providers: [
    {
      provide: APP_ID,
      useValue: 'serverApp'
    }
  ]
```
```
We can propose an alternative to what was deprecated.
Note: It's a mandatory change when migrating to v19, because `withServerTransition` was removed from code in this version.

Reply:
I have problem to understand why would we provide the APP_ID. In fresh ng17 apps there is no such provider.
Does ng19 migration add this provider?

Reply:
It's part of [Angular Migration Guide](https://angular.dev/update-guide?v=18.0-19.0&l=1) which should be adresses manually by customers.
![image](https://github.com/user-attachments/assets/01d9ecb9-62b2-47f5-a2bd-6463077acdeb)
If for some reasons customers need to use `appId` in their app, we can provide equivalent for previous `withServerTransition`
Another option is to not mention this change in our docs at all as they are already addressed in Angular Update Guide to v19 (as well as in [update guide to v16, when the method was deprecated](https://angular.dev/update-guide?v=15.0-16.0&l=3))